### PR TITLE
Ledger Sync - Display error when scanning invalid QR Code

### PR DESCRIPTION
**Pull request by cgrellard-ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 

### 📝 Description

_Replace this text by a clear and concise description of what this pull request is about and why it is needed. Be sure to explain the problem you're addressing and the solution you're proposing._
_For libraries, you can add a code sample of how to use it._
_For bugfixes, you can explain the previous behavior and how it was fixed._
_In case of visual features, please attach screenshots or video recordings to demonstrate the changes._

| Description        | Behavior         |
| ------------- | ------------- |
|Scans a Ledger Sync QR code on desktop with FF disabled on mobile|Screenshot_2024-09-11-17-52-10-82_99c1b3bb631177f8a9e1379ae5b12ffe|
|Scans the old QR code on desktop with FF enabled on mobile|Screenshot_2024-09-11-18-05-16-58_99c1b3bb631177f8a9e1379ae5b12ffe|
|Scans a QR code that has nothing to do with Ledger Sync when FF enabled on mobile|Screenshot_2024-09-11-18-05-35-27_99c1b3bb631177f8a9e1379ae5b12ffe|

### ❓ Context

- **JIRA or GitHub link**: [LIVE-14059] [LIVE-14064] [LIVE-14020]

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-14059]: https://ledgerhq.atlassian.net/browse/LIVE-14059?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/newArch/features/WalletSync/components/Activation/ActivationFlow.tsx
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/newArch/features/WalletSync/components/Activation/ActivationFlow.tsx:1]: You need to add it also on `apps/ledger-live-mobile/src/newArch/features/Accounts/screens/AddAccount/components/StepFlow.tsx`

File: apps/ledger-live-mobile/src/screens/ImportAccounts/Scan.tsx [hunk 1]
```diff
@@ -75,6 +77,13 @@ class Scan extends PureComponent<
           }
         }
       } catch (e) {
+        if (data.match(/host=([0-9A-Fa-f]+)/)) {
+          this.setState({
+            error: new ScannedNewImportQrCode(),
+            progress: 0,
+          });
+          screen("", AnalyticsPage.ScannedIncompatibleApps, { source: "Account Import Sync" });
+        }
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/screens/ImportAccounts/Scan.tsx:86]: Do we have to do it in the catch block? Couldn't we perform the check at the beginning instead?

Comment (cgrellard-ledger) [reply apps/ledger-live-mobile/src/screens/ImportAccounts/Scan.tsx:86]: It's the analytics event for the error that is triggered in the catch (the `setState({ error: new ScannedNewImportQrCode() })` ).
As it's handled by the generic error component I had to trigger the analytics here

**Review by mcayuelas-ledger — CHANGES_REQUESTED** [review]
`StepFlow.tsx` needs to be updated also

**Review by mcayuelas-ledger — APPROVED** [review]

**Pull request merged**